lib/kubernetes-deploy/kubeclient_builder.rb

```diff
@@ -28,7 +29,7 @@ def build_v1beta1_kubeclient(context)
     end
 
     def _build_kubeclient(api_version:, context:, endpoint_path: nil)
-      config = Kubeclient::Config.read(ENV.fetch("KUBECONFIG"))
+      config = GoogleConfig.read(ENV.fetch("KUBECONFIG"))
```

Comment:
I'm not sold about naming because someone reading this code could think that it only supports GKE config. What about `KubeconfigWithGoogleSupport` or whatever?